### DISCUSSION TITLE
How to rearrange overlay order

### DISCUSSION THREAD

**Discussion by P0tt0Man** [runelite/runelite#16467] [Q&A]
Hello.

How do I rearrange or change order of the overlays? Before I had the boost infoboxes by the "wall" of the game inline. To its left, I had the opponents info overlay. I missclicked and screwed up the overlays position, now whenever I try to place them back, no matter what I do, the opponent info gets to the right of the boost infoboxes. I need the infoboxes to be to the right ASAP.

Please help.
![image_2023-04-06_175916335](https://user-images.githubusercontent.com/112856092/230501610-639c33b6-4055-4ec2-be6a-55125ab31d31.png)

**Comment by Nightfirecat** [comment]
There is no way to order overlays within a snap corner. You probably did not have one of those overlays snapped, but manually placed nearby so that it looked like it was in line with the existing snapped overlays.

**Reply by P0tt0Man** [reply]
It was, Im sure of it. Because when I didnt have any boost the opponent overlay and the attack style moved closer to the corner, when I had defensive cast it moved too. The overlays never were one on top of each other.

**Reply by Nightfirecat** [reply]
As I said, the behavior that exists now has not changed.

**Comment by ghost** [comment]
If I can elaborate on OP's issue, because this is also something I'm looking into myself. When overlays are snapped to the same corner/position, they will automatically order themselves, this is what we both expect. However, the order seems to be random or decided by some custom logic that we don't know, and so we'd like to be able to specify the order of overlay elements.

I've often found the snap order when I am re-snapping specific overlay elements to be decided based on the order I enabled those overlays.

Sometimes I can "fix" the order by enabling/disabling the respective overlays until I get what I want - but I can't consistently do it, so I just give up and manually position them.

I think what would be amazing is to be able to decide the order the overlay elements are snapped in AND specify the direction (above, below, left, right of the last snapped element)

**Reply by APKiwi** [reply]
I have this exact same problem and stumbled on the 'work-around' as you did. Did you end up making a request/issue for this?

**Reply by Felanbird** [reply]
@ghost is the filler for deleted Github accounts, so they almost certainly are not going to reply to this question.
